src/services/music-providers/youtube-music/youtube-music.service.ts

```diff
@@ -0,0 +1,286 @@
+            artists: track.artists.map((a) => a.name),
+            source: track,
+        };
+        this.cache.set(searchQuery, result);
+
+        return result;
+    }
+
+    async addTracksToPlaylist(
+        trackIds: string[],
+        playlist: Playlist,
+    ): Promise<void> {
+        await retry<PlaylistLibmuse>(
+            async () => await add_playlist_items(playlist.id, trackIds),
+            async () =>
+                this.notifyThatFunctionCompletedWithError(
+                    'addTracksToPlaylist',
+                ),
+        );
+    }
+
+    async removeTracksFromPlaylist(
+        tracks: Track[],
+        playlist: Playlist,
+    ): Promise<void> {
+        const playlistFromServer: PlaylistLibmuse = await this.getPlaylist(
+            playlist.id,
+            { limit: this.getPlaylistTracksLimit },
+        );
+
+        const trackWithId = tracks.filter((t) => {
+            if (t?.id) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.name} by ${t.artists} was not removed in youtube. Reason id is absent`,
+            );
+        });
+
+        const trackFromServer = playlistFromServer.tracks.filter(
+            (trackFromServer) =>
+                trackWithId.some(
+                    (propsTrack) => propsTrack.id === trackFromServer.videoId,
+                ),
+        );
+
+        const trackFromServerWithSetVideoId = trackFromServer.filter((t) => {
+            if (t.setVideoId) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.title} by ${t.artists} was not removed in youtube. Reason setVideoId is absent`,
+            );
+        });
+
+        if (!trackFromServerWithSetVideoId.length) {
+            return;
+        }
+
+        await this.removePlaylistItemsWrapperOverLibmuseMethod(
+            playlist.id,
+            trackFromServerWithSetVideoId.map((t) => ({
+                videoId: t.videoId,
+                setVideoId: t.setVideoId as string,
+            })),
+        );
+    }
+
+    private artistsToNameArray(artists: SongArtist[]) {
+        let result: string[] = [];
+
+        for (const artist of artists) {
+            if (artist.id) {
+                result.push(artist.name);
+                continue;
+            }
+
+            const names = artist.name.replace(/, & | & | and | и /g, ', ');
+            result = [...result, ...names.split(', ')];
+        }
+
+        return result;
+    }
+
+    private checkThatTrackIsSuitable(
+        findName: string,
+        findArtist: string[],
+        foundName: string,
+        foundArtists: string[],
+    ) {
+        findName = findName.toLowerCase();
+        findArtist = findArtist.map((a) => a.toLowerCase());
+        foundName = foundName.toLowerCase();
+        foundArtists = foundArtists.map((a) => a.toLowerCase());
+
+        const findNameWithoutParticipantDesignations =
```

Comment:
Consider renaming `findNameWithoutParticipantDesignations` and `foundNameWithoutParticipantDesignations` to be more concise and descriptive. For example, `processedFindName` and `processedFoundName` could be alternatives. This will improve code readability, making it easier to understand the function's logic.